common/src/main/java/com/wynntils/models/skillpoint/SavableSkillPointSet.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.skillpoint;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public record SavableSkillPointSet(
+        int strength,
+        int dexterity,
+        int intelligence,
+        int defence,
+        int agility,
+        List<String> armourNames,
+        List<String> accessoryNames,
+        boolean isBuild) {
+    /**
+     * Constructs a new SavableSkillPointSet representing just a loadout.
+     */
+    public SavableSkillPointSet(int[] skillPoints) {
+        this(skillPoints, new ArrayList<>(), new ArrayList<>());
+    }
+
+    /**
+     * Constructs a new SavableSkillPointSet representing a full build with gear.
+     */
+    public SavableSkillPointSet(int[] skillPoints, List<String> armourNames, List<String> accessoryNames) {
+        this(
+                skillPoints[0],
+                skillPoints[1],
+                skillPoints[2],
+                skillPoints[3],
+                skillPoints[4],
+                Collections.unmodifiableList(armourNames),
+                Collections.unmodifiableList(accessoryNames),
+                !armourNames.isEmpty() || !accessoryNames.isEmpty());
+    }
+
+    public int[] getSkillPointsAsArray() {
+        return new int[] {strength, dexterity, intelligence, defence, agility};
+    }
+
+    public int getSkillPointsSum() {
+        return strength + dexterity + intelligence + defence + agility;
+    }
+
+    public int getMinimumCombatLevel() {
+        return (int) Math.ceil(getSkillPointsSum() / 2.0) + 1;
+    }
+
+    public boolean isBuild() {
```

Comment:
This is true if we have any armor or accessories..? Why store it as a boolean?

Reply:
Technically you don't even need this method, since the boolean is already "cached"

Reply:
Sorry, my comments were a bit tarse, I am lying in bed in a strange position and had diffuculty typing. :)

What I meant was that it seemed to me that this value was set to false iff both armor and accessories where empty. But this can as easily be checked on the arrays, and it does not seem to be necessary to cache this in a boolean field; the isBuild() method can check the two arrays directly.